### PR TITLE
feat: integration test suite

### PR DESCRIPTION
**Pull request by waynexia**
## Changes Included

Add a SQL integration test suites and a group of basic SQLs from "get started" document. This test suite only supports local standalone mode now.

To run it, use
```bash
cd tests/runner
cargo run
```

## Notes

The dependency `sqlness` is not published, I'm discussing publish procedure with CeresDB authors (it is derived from there). When things done I'll update it to crates.io registry.

## Known problem

- The start command is `cargo run -- datanode start`, which may cause zombie process when closing. It should be run from the binary directly.

- Due to the `DROP TABLE` clause is not supported now, tests cannot clear its state and will result to "table already exist" when re-run it.

### CODE REVIEW

**Comment by evenyag** [comment]
> Merging [#487](https://codecov.io/gh/GreptimeTeam/greptimedb/pull/487?src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=GreptimeTeam) ([2031a73](https://github.com/GreptimeTeam/greptimedb/commit/2031a7383b59faa1a87526c3e07532903495ac08)) into [develop](https://codecov.io/gh/GreptimeTeam/greptimedb/commit/a179481966c599bdc29330cb56176621815bf30f?el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=GreptimeTeam) ([a179481](https://github.com/GreptimeTeam/greptimedb/commit/a179481966c599bdc29330cb56176621815bf30f)) will **decrease** coverage by `2.22%`.

I think we could ignore this crate in `codecov.yml`.

**Review by evenyag — APPROVED** [review]
LGTM

**Comment by evenyag** [comment]
Maybe better to mention that we need to clean the data directory before runing test.

**Comment by waynexia** [comment]
> Maybe better to mention that we need to clean the data directory before runing test.

I plan to write a document in dev-guide to describe how to run and add integration test when this gets merged. I'll mention this here

File: tests/runner/src/env.rs [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    async fn stop(&self, _mode: &str, mut database: Self::DB) {
+        database.server_process.kill().await.unwrap()
+    }
+}
+
+impl Env {
+    pub async fn start_local() -> GreptimeDB {
+        let server_process = Command::new("cargo")
+            .current_dir("../")
+            .args(["run", "--", "datanode", "start"])
```
Comment (killme2008) [inline tests/runner/src/env.rs:54]: is "datanode" should be changed into "standalone"?

**Review by killme2008 — APPROVED** [review]
LGTM

File: tests/runner/src/env.rs (hunk 1 above)
Comment (waynexia) [reply tests/runner/src/env.rs]: Good catch!

**Pull request merged**